src/main/java/ch/njol/skript/expressions/ExprSubstring.java

```diff
@@ -37,72 +40,92 @@
  * @author Peter Güttinger
  */
 @Name("Subtext")
-@Description("Extracts part of a text. You can either get the first &lt;x&gt; characters, the last &lt;x&gt; characters, or the characters between indices &lt;x&gt; and &lt;y&gt;."
+@Description("Extracts part of a text. You can either get the first &lt;x&gt; characters, the last &lt;x&gt; characters, the characters at certain indexes &lt;x&gt;, or the characters between indices &lt;x&gt; and &lt;y&gt;."
 		+ " The indices &lt;x&gt; and &lt;y&gt; should be between 1 and the <a href='#ExprLength'>length</a> of the text (other values will be fit into this range).")
-@Examples({"set {_s} to the first 5 characters of the text argument"
-		, "message \"%subtext of {_s} from characters 2 to (the length of {_s} - 1)%\" # removes the first and last character from {_s} and sends it to the player or console"})
-@Since("2.1")
+@Examples({"set {_s} to the first 5 characters of the text argument",
+		"message \"%subtext of {_s} from characters 2 to (the length of {_s} - 1)%\" # removes the first and last character from {_s} and sends it to the player or console",
+		"set {_characters::*} to characters at 1, 2 and 7 in player's display name",
+		"send the last character of all players' names"})
+@Since("2.1, INSERT VERSION (character at, multiple strings support)")
 public class ExprSubstring extends SimpleExpression<String> {
 	static {
 		Skript.registerExpression(ExprSubstring.class, String.class, ExpressionType.COMBINED,
 				"[the] (part|sub[ ](text|string)) of %strings% (between|from) (ind(ex|ices)|character[s]|) %number% (and|to) (index|character|) %number%",
-				"[the] (1¦first|2¦last) [%-number%] character[s] of %strings%", "[the] %number% (1¦first|2¦last) characters of %strings%");
+				"[the] (1¦first|2¦last) [%-number%] character[s] of %strings%", "[the] %number% (1¦first|2¦last) characters of %strings%",
+				"[the] character[s] at [(index|position|indexes|indices|positions)] %numbers% (in|of) %strings%");
 	}
 	
 	@SuppressWarnings("null")
 	private Expression<String> string;
 	@Nullable
 	private Expression<Number> start, end;
+	private int pattern;
 	
 	@SuppressWarnings({"unchecked", "null"})
 	@Override
 	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
-		string = (Expression<String>) exprs[parseResult.mark == 0 ? 0 : 1];
-		start = parseResult.mark == 1 ? null : exprs[parseResult.mark == 0 ? 1 : 0] == null ? new SimpleLiteral<>(1, false) : (Expression<Number>) exprs[parseResult.mark == 0 ? 1 : 0];
-		end = parseResult.mark == 2 ? null : exprs[parseResult.mark == 0 ? 2 : 0] == null ? new SimpleLiteral<>(1, false) : (Expression<Number>) exprs[parseResult.mark == 0 ? 2 : 0];
+		pattern = matchedPattern;
```

Comment:
instead of storing the pattern int, can you store a boolean like `boolean usedCharAtPattern = matchedPattern == 0;`? after that, just change all your pattern equality checks to use the boolean instead and then this PR is golden <3

Reply:
Done 👍